pkg/util/testutil/mockbackend.go

```diff
@@ -0,0 +1,24 @@
+package testutil
+
+import (
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend"
+	cmdBackend "github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/backend"
+)
+
+// MockBackendInstance sets the backend instance for the test and cleans it up after.
+func MockBackendInstance(t *testing.T, b backend.Backend) {
```

Comment:
There's a couple more places where we have an identical function. I'll follow up with a clean up PR.